lib/native_service/io_file_storage_delegate.dart

```
@@ -47,11 +48,62 @@ final class IoFileStorageDelegate extends FileStorageDelegate {
   }
 
   @override
-  Future<bool> requestWriteExternalStoragePermission() async {
+  Future<bool> requestExternalStoragePermission({bool read = false, bool write = false}) async {
     final bool? result = await methodChannel.invokeMethod<bool>(
```

This should throw an UnsupportedError on platforms other than Android.